cmake/HPX_AddExecutable.cmake

```diff
@@ -149,7 +149,7 @@ function(add_hpx_executable name)
     endif()
   endif()
 
-  set_target_properties(${name}_exe PROPERTIES OUTPUT_NAME ${name})
+  set_target_properties(${name}_exe PROPERTIES OUTPUT_NAME "${HPX_OUTPUT_PREFIX}${name}")
```

Comment:
Is this meant for prefixing e.g. examples with `hpx_` when installing them systemwide? And meant to be set by the user? Would you mind adding it as an option somewhere around [here](https://github.com/STEllAR-GROUP/hpx/blob/c1478e81321c67f55ca029640b75582c1cb70231/CMakeLists.txt#L216-L248)?

Reply:
Done.

Reply:
Thanks! Could you actually make it `HPX_WITH_OUTPUT_PREFIX` (Is `HPX_WITH_EXECUTABLE_PREFIX` clearer? That's what you write in the description.) instead to be consistent with other options. Sorry, should've made that clear the first time around!

Reply:
Done.

Reply:
Perfect, thanks!